### ISSUE TITLE
Language Bar icon incorrectly dislpayed while using Google Chrome

### ISSUE THREAD

**Issue by Rohupt** [saschanaz/ime-rs#5]
The Language Bar icon says that the IME is disabled when using in Google Chrome, but it's actually working.
![image](https://user-images.githubusercontent.com/34348907/158331694-799ceab9-64b4-498e-b1cb-009853814046.png)

**Comment by saschanaz** [comment]
That doesn't happen here 🤔 Does it happen on any other app?

**Comment by saschanaz** [comment]
(I'll only fix a regression, I won't fix anything if a bug was already there in the demo app)

**Comment by saschanaz** [comment]
It doesn't happen at all on Chrome, vscode, nor Word on my machine, so I have no advice...

**Comment by saschanaz** [comment]
Perhaps check if your apps are all x64? 🤔

**Comment by Rohupt** [comment]
Yeah, it doesn't work at all on 32-bit applications (except Teams), and falsely "disabled" in some of the 64-bit ones (Chrome, Word,...).

In the demo project it only worked on Metro apps but not desktop apps. What did you change for it to work on desktop apps?

**Comment by saschanaz** [comment]
Hmm? The demo project has no behavioral difference with the current port, again on my machine 🤔 I don't intend to do any "fix" here, so I didn't do anything about it.

**Comment by Rohupt** [comment]
The original documentation says this:

> This implementation works only for Windows Store apps that use the built-in edit controls. This implementation won't work if the app uses the custom edit control from the subset of Text Service Framework (TSF) APIs available in the Windows Runtime.

And it indeed doesn't work on anything but Windows Store apps on my machine, so I just assumed it was true. But this one does work on some of the desktop apps.

I'm starting to think that it's my machine that's the problem, but do you happen to have any idea about it?

**Comment by Rohupt** [comment]
> You mean you see the same behavior with the original demo? I'm not sure, all I know at this point is that GUID_COMPARTMENT_KEYBOARD_DISABLED is related, which is used in `CSampleIME::_IsKeyboardDisabled`. I haven't touched that part, so I barely knows anything there.

So I removed the 2 lines that use it, and the mode indicator appeared again, and it started to work on File Explorer. The same thing still didn't apply with the original demo though.

My machine started to act a bit funny (Explorer and Chrome each suddenly terminated once when I was changing the mode/language), but I guess it's solved? (Probably you should put a line or two into the readme.)

**Comment by Rohupt** [comment]
I removed lines 229 and 230 of KeyEventSink.cpp:
```
        CCompartment CompartmentKeyboardDisabled(_pThreadMgr, _tfClientId, GUID_COMPARTMENT_KEYBOARD_DISABLED);
        CompartmentKeyboardDisabled._GetCompartmentBOOL(isDisabled);
```

And yes, when I removed those lines from the original demo I didn't see any change.

But on another note, even one of the IMEs that had been working well until recently started to act up the same way as this issue, so maybe my Windows is corrupted somewhere.

**Comment by Rohupt** [comment]
> But on another note, even one of the IMEs that had been working well until recently started to act up the same way as this issue, so maybe my Windows is corrupted somewhere.

I have to readdress this, but after installing this both the Japanese and Korean IMEs in my machine starts acting up. The secondary typing mode button (the A/あ and A/가 buttons) all replaced by this one's (originally 英/中).
